spec/generators/routes_generator_spec.rb

```diff
@@ -120,6 +115,7 @@ class ModelWithoutDBTable < ApplicationRecord; end
 
     run_generator
 
-    expect(routes).to contain('root to: "customers#index')
+    expect(routes).to contain('root to: "application#show"')
+    # Why does the root get changed when namespace is included..
```

Comment:
It's actually not changing. Instead, this refers to an additional `root` line in the file, which was there before this PR. The one that we were originally testing for (`customers#index`) is not there any more, so you have found that this other one is there and changed the expectation.

This example runs the generator against an existing, mostly empty routes file. If you search for `application#show` in the codebase, you'll find it:

https://github.com/thoughtbot/administrate/blob/84084235463a6761f25108382cb81c13632d2052/spec/app_templates/config/routes.rb#L1-L3

So that `application#show` is there already, is correct, and shouldn't change when the generator runs. That's fine. When the generator runs, new lines are added. If you insert a `puts File.read(routes)` right after `run_generator`, you'll see what the routes file actually looks like by the time the expectation is checked. It looks like this:

```ruby
Rails.application.routes.draw do
  namespace :admin do
    namespace :active_record do
      resources :internal_metadata
    end
    namespace :active_record do
      resources :schema_migrations
    end
    namespace :primary do
      resources :schema_migrations
    end
    resources :customers
    resources :orders
    namespace :blog do
      resources :posts
    end
    resources :countries
    resources :line_items
    resources :log_entries
    resources :pages
    resources :payments
    resources :products
    resources :product_meta_tags
    resources :series
    
    root to: "active_record/internal_metadata#index"
  end
  root to: "application#show"
end
```

So from here you can see that the problem is more with new namespaced that are being added and probably shouldn't: `active_record` and `primary`. The root is assigned arbitrarily to the `#index` of the first model in the list, which happens to be `active_record/internal_metadata`.

I think that this spec example should change so that the expectation is a bit more strict about what's in the file. Perhaps it should match a full file, instead of searching for a specific line.

Reply:
Thank you for the information here @pablobm,

`puts File.read(routes)` is a helpful visibility idea - slightly annoyed I didn't see it. 

Yeah those namespaces shouldn't be used - so that's helpful to know what to investigate. Also testing the whole file makes more sense.